code-challenges.test.js

```diff
@@ -16,35 +17,95 @@
 // a) Create a test with an expect statement using the variable provided. 
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+describe("mixIt", ()=>{
+    it("Function that takes in an array, removes the first item from the array and shuffles the remaining content.", ()=>{
+        const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+        output1 = ["yellow", "blue", "pink", "green"]
+        const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+        output2 = ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+        expect(mixIt(colors1)).toEqual(expect.arrayContaining(output1))
+        expect(mixIt(colors1)[0]).not.toEqual("yellow")
+        expect(mixIt(colors2)).toEqual(expect.arrayContaining(output2))
+        expect(mixIt(colors2)[0]).not.toEqual("saffron")
+    })
+})
 
-// b) Create the function that makes the test pass.
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
 
+// b) Create the function that makes the test pass.
+// The function takes in an array
+const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+const mixIt = (array) => {
+    // we use the slice accessor method to remove the first element in the array. Array.slice will start the array at the given index and continue until the index of the second given argument. If no second argument is given it will continue till the end of the array.
+   let newArr = array.slice(1)
+   // creating the condition to count down so the while loop stops.
+   let index = newArr.length
+   // initializing a variable to hold a random number
+   let random
+   // the while loop will continue until the index = 0, since 0 is falsy we don't need to actually put the != part
+   while (index) {
```

Comment:
This is wonderfully explained, great job!  If you want to stretch yourself, consider how you can solve this using higher order functions.